mqt/syrec/syrec_editor.py

```diff
@@ -278,122 +345,133 @@ def sim(self):
 
         self.writeEditorContentsToFile()
 
-        prog = program()
+        if self.buildType== [0,0]:
 
-        error_string = prog.read("/tmp/out.src")
-        if error_string == "PARSE_STRING_FAILED":
             if self.parser_failed is not None:
-                self.parser_failed("Editor is Empty")
-            return
-        elif error_string != "":
-            if self.build_failed is not None:
-                self.build_failed(error_string)
+                self.parser_failed("Sythesizer not selected")
             return
 
-        circ = circuit()
+        else:
+
+            prog = program()
+
+            error_string = prog.read("/tmp/out.src")
+            if error_string == "PARSE_STRING_FAILED":
+                if self.parser_failed is not None:
+                    self.parser_failed("Editor is Empty")
+                return
+            elif error_string != "":
+                if self.build_failed is not None:
+                    self.build_failed(error_string)
+                return
```

Comment:
the same as above goes here. if the circuit has not been synthesized, there is nothing to simulate.
the simulation routine should not duplicate all the code of the other functions.

Probably, the best solution would be to disable the `stat` and `simulation` buttons until a circuit has been synthesized.